public/pages/ReviewAndCreate/components/DetectorDefinitionFields/DetectorDefinitionFields.tsx

```diff
@@ -51,6 +66,64 @@ export const DetectorDefinitionFields = (
     ),
   };
 
+  const handleCalloutGeneralLogic = () => {
+    //When validation response is loading then displaying loading spinner, don't display
+    // after clicking on "create detector" button as isLoading will be true from that request
+    if (props.isLoading && !props.isCreatingDetector) {
+      return (
+        <EuiCallOut
+          title={
+            <div>
+              <EuiFlexGroup direction="row" gutterSize="xs">
+                <EuiLoadingSpinner size="l" style={{ marginRight: '12px' }} />
+                <EuiText>
+                  <p>Validating detector configurations</p>
+                </EuiText>
+              </EuiFlexGroup>
+            </div>
+          }
+          style={{ marginBottom: '10px' }}
+          size="s"
+          color="primary"
+        />
+      );
+    }
+    // if validationResponse is not undefined and there was no error from validation display one
+    // of the callout options.
+    if (props.validationResponse != undefined && !props.validationError) {
+      if (props.validDetectorSettings) {
+        return (
+          <EuiCallOut
+            title="Detector settings are validated"
+            color="success"
+            iconType="check"
+            size="s"
+            style={{ marginBottom: '10px' }}
+          ></EuiCallOut>
+        );
+      } else if (
+        !props.validDetectorSettings &&
+        props.validationResponse.hasOwnProperty('message')
+      ) {
+        return (
+          <EuiCallOut
+            title="Issues found in the detector settings"
+            color="danger"
+            iconType="alert"
+            size="s"
+            style={{ marginBottom: '10px' }}
+          >
```

Comment:
I guess naming is confusing but validationError is regarding any exception from validation API such as security exception is caught which should lead to no validation callout at all. So when !props.validationError is true it means props.validationError is false cause no exception was caught. props.validDetectorSettings refers to the actual ValidationAPI response and if detector settings are valid or not (data filter parsing issue and etc.) So basically if there is no exception from validate API call then validDetectorSettings can still be true or false depending on validation API response.

Reply:
> validationError is regarding any exception from validation API such as security exception is caught which should lead to no validation callout at all.

Is there a specific reason why not? Could be helpful to see any error here, since I'm assuming it would instantly fail when the user tries to click 'Create detector' anyways, if it is some security issue or something else.

I'm ok to leave as-is given the time, but can you add comments explaining the different names and what they mean? And/or rename the vars?

Reply:
My main logic behind this was to only use the callouts for the purpose of what the validation API is meant for. For validating against the list of checks that are blocker issues. I didn't want there to be a weird callout that I wasn't prepared for incase there are other types of exceptions that occur that I didn't think of. However I think it also totally makes sense to display security issues or other exceptions as part of validation call out but as of now I left it as the same UI as other exceptions across AD plugin. It will show up as notification on bottom right like other exceptions show. I'll add comments to explain names better and revisit this for 1.3.

Reply:
Not sure if done addressing comments, but can you add some comments like you've mentioned here? Or change the names?